src/components/actions/executeAction.ts

```diff
@@ -25,10 +25,10 @@ export async function executeAction(
   payload: ActionPayload,
   actionLibrary: ActionLibrary,
   additionalObjects: any = {},
-  previewData?: Partial<ActionApplicationData>
+  applicationDataExtra?: Partial<ActionApplicationData>
```

Comment:
Not really related to this PR (was going to be initially), but makes sense to give this field a more generic name since it could be used by more than just the "preview" endpoint. (Have kept the name "previewData" *within* the "preview" endpoint, it just passes on the more generic name to the processTrigger function.